### DISCUSSION TITLE
Serverless Offline Requires AWS Credentials for Development

### DISCUSSION THREAD

**Discussion by logicielgurwinder** [serverless/serverless#12618] [Q&A]
Hi,

I recently installed Serverless Framework version 4 and I'm using serverless-offline for local development. However, when I run the serverless offline command, it prompts me for AWS credentials. Since we are only doing local development and don't have an AWS account, I'm not sure why it's asking for these credentials.

Is there a way to configure Serverless or serverless-offline to run without requiring AWS credentials for local development? Any guidance or tips would be greatly appreciated.

Thank you!

**Comment by OdugbesanOpeoluwa** [comment]
I don't think serverless-offline for local development is supported for  Serverless Framework version 4

**Comment by ronkot** [comment]
serverless-offline actually supports Serverless v4. We've been using this combination successfully for two months, but today, our local Serverless-offline development suddenly started requiring AWS credentials, failing with the error: `AWS credentials missing or invalid. Original error from AWS: Could not load credentials from any providers`. We're still trying to figure out why this issue occurred. I checked and confirmed that this does not happen with Serverless v3.

**Comment by prateekbhaskarliquiloans** [comment]
This is not yet resolved.
Serverless offline should not try to connect AWS.
Can someone please look into this on priority?
Local development is stalled due to this.

**Comment by ronkot** [comment]
The reason for this seems to be that serverless framework has added a feature that automatically stores some application state to aws s3 bucket. This feature was added in some recent minor version update but it broke our local dev (as well as deployments) because of missing credentials and/or credential permissions.

The solution is to downgrade your serverless framekwork to some earlier version that is not using s3 bucket state and pin it down. We're using 4.2.3 and at least it is safe to use. **NOTE!** Serverless framework has also implemented its own way to handle version updates. It doesn't follow the lockfile versions as one would expect, but instead  `frameworkVersion` version that is defined in serverless.yml. So to pin down the excact version you have to set it to serverless.yml.  See https://github.com/serverless/serverless/issues/12886 for more info.

(I don't like the way serverless framework is evolving with its own dependency managament system etc. I'm planning to substitute it with some alternative like SAM.)
